src/test/engineeffectsdelay_test.cpp

```diff
@@ -0,0 +1,313 @@
+// Tests for engineeffectsdelay.cpp
+
+// Premise: internal Mixxx structure works with a stereo signal.
+// If the mixxx::kEngineChannelCount wouldn't be a stereo in the future,
+// tests have to be updated.
+
+#include "engine/effects/engineeffectsdelay.h"
+
+#include <gtest/gtest.h>
+
+#include <QTest>
+#include <QtDebug>
+
+#include "engine/engine.h"
+#include "test/mixxxtest.h"
+#include "util/sample.h"
+#include "util/types.h"
+
+namespace {
+
+static_assert(mixxx::kEngineChannelCount == mixxx::audio::ChannelCount::stereo(),
+        "EngineEffectsDelayTest requires stereo input signal.");
+
+class EngineEffectsDelayTest : public MixxxTest {
+  protected:
+    void AssertIdenticalBufferEquals(const CSAMPLE* pBuffer,
+            int iBufferLen,
+            CSAMPLE* pReferenceBuffer,
+            int iReferenceBufferLength) {
+        EXPECT_EQ(iBufferLen, iReferenceBufferLength);
+
+        for (int i = 0; i < iBufferLen; ++i) {
+            EXPECT_FLOAT_EQ(pBuffer[i], pReferenceBuffer[i]);
+        }
+    }
+
+    EngineEffectsDelay m_effectsDelay;
+};
+
+TEST_F(EngineEffectsDelayTest, NegativeDelayValue) {
+    const SINT numSamples = 4;
+
+    // Set negative delay value.
+    m_effectsDelay.setDelayFrames(-1);
+
+    CSAMPLE inputBuffer[] = {-100.0, 100.0, -99.0, 99.0};
+
+    CSAMPLE* pOutput = SampleUtil::alloc(numSamples);
+
+#ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
+    // Set thread safe for EXPECT_DEATH.
+    GTEST_FLAG_SET(death_test_style, "threadsafe");
```

Comment:
At least it would be a new pattern in out test code.

I think it is a kind of paranoia to test test code, on the other hand it is a fixture that guarantees that a certain assertion is not removed and the code is already written. 
I have no strong opinion here. Maybe keep the code but not invest time to introduce similar test elsewhere.

Reply:
Well, if this was Rust code with explicit error types, or even C++ with proper exception use, I'd be in favor of testing failure modes. But in mixxx, our error handling is almost non-existent. We just have our `DEBUG_ASSERT`s so we're not flying completely blind in debug builds and in release builds we're just praying nothing will happen. I don't know if it makes sense to test `DEBUG_ASSERT`s which are already a hack in the first place.

Since the code is already written, lets just keep it as long as the test don't turn out to be flaky.

Reply:
Okay, I understand your reason for the question if it is a good idea. So, as you wrote, for now, I will keep this code as-is. On the other hand, it isn't a problem to change it.